src/logworker.cpp

```diff
@@ -29,6 +29,8 @@ namespace g3 {
 
       if (_sinks.empty()) {
          std::string err_msg {"g3logworker has no sinks. Message: ["};
+
+         // TODO: following line will cause seg fault if empty msgPtr is provided
```

Comment:
I don't think the TODO is needed. Here is no way for a msg to be nullptr except through unit tests